cpp/include/raft/linalg/rsvd.cuh

```diff
@@ -255,15 +412,188 @@ void rsvd_perc(
            V.value().data_handle(),
            PC_perc,
            UpS_perc,
-           use_bbt,
+           false,
            U.has_value(),
            V.has_value(),
-           use_jacobi,
+           false,
            tol,
            max_sweeps,
            handle.get_stream());
 }
 
+/**
+ * @brief randomized singular value decomposition (RSVD) on a column major
+ * rectangular matrix using symmetric Eigen decomposition, by specifying the PC and upsampling
+ * ratio. The rectangular input matrix is made square and symmetric using B @ B^T
+ * @param[in] handle raft::handle_t
+ * @param[in] M input raft::device_matrix_view with layout raft::col_major of shape (M, N)
+ * @param[out] S_vec singular values raft::device_vector_view of shape (K)
+ * @param[in] PC_perc percentage of singular values to be computed
+ * @param[in] UpS_perc upsampling percentage
+ * @param[in] tol tolerance for Jacobi-based solvers
+ * @param[in] max_sweeps maximum number of sweeps for Jacobi-based solvers
+ * @param[out] U optional left singular values of raft::device_matrix_view with layout
+ * raft::col_major
+ * @param[out] V optional right singular values of raft::device_matrix_view with layout
+ * raft::col_major
+ */
+template <typename ValueType, typename IndexType>
+void rsvd_perc_symmetric(
+  const raft::handle_t& handle,
+  raft::device_matrix_view<ValueType, IndexType, raft::col_major> M,
+  raft::device_vector_view<ValueType, IndexType> S_vec,
+  ValueType PC_perc,
+  ValueType UpS_perc,
+  ValueType tol,
+  int max_sweeps,
+  std::optional<raft::device_matrix_view<ValueType, IndexType, raft::col_major>> U = std::nullopt,
```

Comment:
Note that if users provide `nullopt` explicitly as an argument, sometimes the compiler has trouble deducing types.  It can be worth adding a test to make sure that compiles.  If not, there are a couple different ways to fix it.  My PR #830 takes one approach that can handle multiple `optional` arguments.

Reply:
Doesn't seem like we are deducing the type here though, are we? I have explicitly specified which type is expected

Reply:
@divyegala Surprisingly, this doesn't always work.  It could be worth writing a little test that passes in `std::nullopt`.  If it compiles, you're fine.

Reply:
Understood! Yes, it does fail the compilation. On that note, I figured out that your solution is a little bit more complicated than necessary. The aliases and type-traits are not required because `std::nullopt` arguments will always go in the overloaded function. From there on, you can directly construct the optional and call the first overload.

Reply:
@divyegala wrote:
> On that note, I figured out that your solution is a little bit more complicated than necessary. The aliases and type-traits are not required because std::nullopt arguments will always go in the overloaded function. 

The benefit of putting the optional arguments at the end and defaulting them to `nullopt` : - )